src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py

```diff
@@ -493,6 +494,198 @@ def test_azconfig_import_export_naming_conventions(self, resource_group, locatio
+                                       self.check('[1].key', entry_key),
+                                       self.check('[1].value', updated_value),
+                                       self.check('[1].label', dest_label)]).get_output_in_json()
+        assert len(deleted_kvs) == 2
+
+        # Not specifying a label or preserve-labels should assign null label and export only one KV and one feature
+        self.cmd(
+            'appconfig kv export --connection-string {src_connection_string} -d {import_source} --dest-connection-string {dest_connection_string} --label {src_label} -y')
+        deleted_kvs = self.cmd('appconfig kv delete --connection-string {dest_connection_string} --key {key} --label {label} -y',
+                               checks=[self.check('[0].key', internal_feature_key),
+                                       self.check('[0].label', null_label),
+                                       self.check('[1].key', entry_key),
+                                       self.check('[1].value', updated_value),
+                                       self.check('[1].label', null_label)]).get_output_in_json()
+        assert len(deleted_kvs) == 2
+
+        # Preserving labels and exporting all kv and all features
+        self.cmd(
+            'appconfig kv export --connection-string {src_connection_string} -d {import_source} --dest-connection-string {dest_connection_string} --label {src_label} --preserve-labels -y')
+        deleted_kvs = self.cmd('appconfig kv delete --connection-string {dest_connection_string} --key {key} --label {label} -y',
+                               checks=[self.check('[0].key', internal_feature_key),
+                                       self.check('[0].label', entry_label),
+                                       self.check('[1].key', internal_feature_key),
+                                       self.check('[1].label', updated_label),
+                                       self.check('[2].key', entry_key),
+                                       self.check('[2].value', entry_value),
+                                       self.check('[2].label', entry_label),
+                                       self.check('[3].key', entry_key),
+                                       self.check('[3].value', updated_value),
+                                       self.check('[3].label', updated_label)]).get_output_in_json()
+        assert len(deleted_kvs) == 4
+
+        # Error when both dest-label and preserve-labels is specified
+        self.kwargs.update({
+            'label': dest_label
+        })
+        with self.assertRaisesRegexp(CLIError, "Export failed! Please provide only one of these arguments: 'dest-label' or 'preserve-labels'."):
+            self.cmd('appconfig kv export --connection-string {src_connection_string} -d {import_source} --dest-connection-string {dest_connection_string} --label {src_label} --dest-label {label} --preserve-labels -y')
```

Comment:
from my understanding, your --preserve-labels argument is in boolean type. Why do you input -y here? Do you want to use the confirmation type?

Reply:
-y is for confirmation. AFAIK '--preserve-labels' is the same as '--preserve-labels true'.

---
In reply to: [365276415](https://github.com/Azure/azure-cli/pull/11756#discussion_r365276415) [](ancestors = 365276415)